pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -961,7 +966,7 @@ func (nc *Controller) tryUpdateNodeHealth(ctx context.Context, node *v1.Node) (t
 			}
 		}
 		// We need to update currentReadyCondition due to its value potentially changed.
-		_, currentReadyCondition = controllerutil.GetNodeCondition(&node.Status, v1.NodeReady)
+		currentReadyCondition = controllerutil.GetNodeCondition(&node.Status, v1.NodeReady)
```

Comment:
Would it make sense to check if `currentReadyCondition` is not nil here?

Reply:
the `currentReadyCondition ` has been used to tree way below, it will not cause panic.

https://github.com/kubernetes/kubernetes/blob/3aa8c59fec0bf339e67ca80ea7905c817baeca85/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L966

https://github.com/kubernetes/kubernetes/blob/3aa8c59fec0bf339e67ca80ea7905c817baeca85/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L722

https://github.com/kubernetes/kubernetes/blob/3aa8c59fec0bf339e67ca80ea7905c817baeca85/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L726

Reply:
if currentReadyCondition is nil, the node status.conditions[].ready will be set

https://github.com/kubernetes/kubernetes/blob/3aa8c59fec0bf339e67ca80ea7905c817baeca85/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L930-L964

Reply:
ok, would it make sense to check if currentReadyCondition is not nil here? What I mean that the return value of `controllerutil.GetNodeCondition` is checked in all other places where it's called, but not in this one.

Reply:
the `controllerutil.GetNodeCondition` return nil have been checked in all places.
And I have try to set `controllerutil.GetNodeCondition` return nil always, the unit test been pass. 
```golang
func GetNodeCondition(status *v1.NodeStatus, conditionType v1.NodeConditionType) *v1.NodeCondition {
	return nil
}
```

Reply:
I'm not sure I follow. This is an example illustrating what I was trying to say:
In all places except this one there is an explicit check for return value right after the function call, e.g.
```
condition := controllerutil.GetNodeCondition(&node.Status, v1.NodeReady)
if condition == nil {
```

However, the return value is not checked after the call on the line 969:
```
currentReadyCondition = controllerutil.GetNodeCondition(&node.Status, v1.NodeReady)
// currentReadyCondition is not checked here
if !apiequality.Semantic.DeepEqual(currentReadyCondition, &observedReadyCondition) {
```

Reply:
The node status has been assigned in line 935-968 before this place. so that the currentReadyCondition on the line 969 won't be nil.
```golang
currentCondition := controllerutil.GetNodeCondition(&node.Status, nodeConditionType) 
if currentCondition == nil { 
    logger.V(2).Info("Condition of node was never updated by kubelet", "nodeConditionType", nodeConditionType, "node", klog.KObj(node)) 
    node.Status.Conditions = append(node.Status.Conditions, v1.NodeCondition{ 
    Type:               nodeConditionType, 
    …… 
}else{
// line 969, the node status conditions type ready is not nil
currentReadyCondition = controllerutil.GetNodeCondition(&node.Status, v1.NodeReady) 
```
Also it is clear that check and assign like this:
```golang
if currentReadyCondition == nil {
        currentReadyCondition = &v1.NodeCondition{
		Type:               nodeConditionType,
		Status:             v1.ConditionUnknown,
		Reason:             "NodeStatusNeverUpdated",
		Message:            "Kubelet never posted node status.",
		LastHeartbeatTime:  node.CreationTimestamp,
		LastTransitionTime: nowTimestamp,
	}
}
```

Reply:
I add an explicit check. please review again.

Reply:
Thanks. I'm ok with this. Let other reviewers to review it.